src/core/capture_loop.py

```
@@ -145,9 +201,20 @@ def _check_end_conditions(self) -> bool:
-        # 重複検出チェック
-        is_duplicate = self.capture_manager.check_end_condition()
-        return self.check_duplicate_end(is_duplicate)
+        # 重複検出チェック（最後にキャプチャした画像で判定）
+        if self.captured_images:
+            # 最後のキャプチャ画像を読み込んで重複チェック
+            from PIL import Image
+            import numpy as np
+            
+            last_image_path = self.captured_images[-1]
+            with Image.open(last_image_path) as img:
+                img_array = np.array(img)
```

Ruff (0.12.2)

Local variable `img_array` is assigned to but never used

Remove assignment to unused variable `img_array`

(F841)